File: backend/src/main/java/com/programmers/pcquotation/domain/estimate/repository/EstimateRepository.java
package com.programmers.pcquotation.domain.estimate.repository;

import java.util.List;

import org.springframework.data.jpa.repository.JpaRepository;

import com.programmers.pcquotation.domain.estimate.entity.Estimate;
import com.programmers.pcquotation.domain.seller.entitiy.Seller;

import jakarta.validation.constraints.NotNull;

public interface EstimateRepository extends JpaRepository<Estimate, Integer> {
	List<Estimate> getAllByEstimateRequest_Id(Integer estimateRequestId);

	List<Estimate> getAllBySeller(@NotNull Seller seller);

	Estimate getEstimateById(Integer id);
}

File: backend/src/main/java/com/programmers/pcquotation/domain/estimate/repository/EstimateRepository.kt
package com.programmers.pcquotation.domain.estimate.repository

import com.programmers.pcquotation.domain.estimate.entity.Estimate
import com.programmers.pcquotation.domain.seller.entity.Seller
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.data.jpa.repository.JpaRepository

interface EstimateRepository : JpaRepository<Estimate, Int> {
    fun getAllByEstimateRequestId(estimateRequestId: Int): List<Estimate>

    fun getAllBySeller(seller: Seller): List<Estimate>

    fun findAllBySeller(seller: Seller, pageable: Pageable): Page<Estimate>

    fun getEstimateById(id: Int): Estimate
}